crates/ruff_linter/src/rules/pydocstyle/rules/ends_with_punctuation.rs

```diff
@@ -103,7 +107,7 @@ pub(crate) fn ends_with_punctuation(checker: &mut Checker, docstring: &Docstring
         if !trimmed.ends_with(['.', '!', '?']) {
             let mut diagnostic = Diagnostic::new(EndsInPunctuation, docstring.range());
             // Best-effort fix: avoid adding a period after other punctuation marks.
-            if !trimmed.ends_with([':', ';']) {
+            if !trimmed.ends_with([':', ';', '?', '!']) {
```

Comment:
Looking at the code again, I think this change is unnecessary because line 107 already checks that the text doesn't end with `!` or `?`

Reply:
Agreed.